### ISSUE TITLE
Species Transport

### ISSUE THREAD

**Issue by PetroleumEngineer** [ansys/pyfluent#2016]
### 🔍 Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### 🐞 Description of the bug

Hello,
I started working on the Ansys fluent model. In my model, I am working with a species transport model. I am using a new tool that was released with 2023R2 Ansys fluent, called Water Chemistry & Corrosion mechanism.

I am trying to activate this tool in Python however I can't locate it at all. I am using  VsCode with Python. Everything works perfectly, the only problem is that I can not find it.

![image](https://github.com/ansys/pyfluent/assets/86235779/1b9437ae-b661-478e-9e21-c8e6ab311c5d)
![image](https://github.com/ansys/pyfluent/assets/86235779/7279db07-7029-41ec-b05c-5edf7a2b9930)
![image](https://github.com/ansys/pyfluent/assets/86235779/941027b8-fd9b-4530-8aec-ccb263a77823)

Please help m,e to figure this out

### 📝 Steps to reproduce

described above

### 💻 Which operating system are you using?

Windows

### 📀 Which ANSYS version are you using?

2023R2

### 🐍 Which Python version are you using?

3.10

### 📦 Installed packages

```shell
no
```

**Comment by seanpearsonuk** [comment]
@dnwillia @dnwillia-work it seems not.
@h-krishnan @gyeole might like to confirm.

**Comment by gyeole** [comment]
No it is not yet covered in Settings APIs

**Comment by seanpearsonuk** [comment]
Thanks very much, @gyeole

**Comment by dnwillia** [comment]
I guess we can mark it as an enhancement then, since it's not yet supported by the `setup` API.  It is there for the `tui`

**Comment by dnwillia-work** [comment]
I think the error is because  `volumetric_species` is a named object like `mixture`.  So, you need to give a name for the species you want to access:
```
solver.setup.materials.mixture['mixture-template'].species.volumetric_species['h2o']
```
The method `get_object_names()` is just telling you which ones are valid so that you can access and edit them.

I must admit I've not set up a species case from scratch using the API, we really need a better example for this.  That said, I would think you should be able to add new materials there by simply doing something like this:
```
solver.setup.materials.mixture['mixture-template'].species.volumetric_species['co32-'] = { ... }
```
and pass one of the dictionaries you echoed in an earlier post for the material `co32-`.  I'll try it when I get a chance and repost here if I get it working.

**Comment by PetroleumEngineer** [comment]
Hi, I tried your method before but it did not work. Instead, I wrote functions that create the species I want. Then I am adding them to the mixture ei want. and after that, I am deleting species. (I don't need them because they are already in the mixture)

solver.setup.materials.fluid['co32-<l>'] = {                    # Manual water chemistry setup
        'name': 'co32-<l>',
        'chemical_formula': 'co32-<l>',
        'density': {'option': 'constant', 'value': 988},
        'molecular_weight': {'option': 'constant', 'value': 60},
        'formation_entropy': {'option': 'constant', 'value': 69902.211},
        'formation_enthalpy': {'option': 'constant', 'value': -285841220.0},
        'reference_temperature': {'option': 'constant', 'value': 298},
        'speed_of_sound': {'option': 'none'},
        'charge': {'option': 'constant', 'value': -2}}
        
I just have A BIG problem, for some reason neither TUI nor line commands allow me to set up the backward reactions for species (Arrhenius backward rate). This is a crucial part of my research. Am I mistaken or at this moment we can not set backward reactions through python?
![image](https://github.com/ansys/pyfluent/assets/86235779/a24b0c37-6949-4d81-a30f-496085664dd1)

**Comment by PetroleumEngineer** [comment]
any updates? I really need help to know how I can access setting up (Arrhenius backward rates)

**Comment by dnwillia-work** [comment]
Yeah, I'd been looking at the mixture setup a little last week and ran into some trouble so still WIP there.  I would think that setting should be in the TUI, I'll check.

**Comment by gyeole** [comment]
In TUI it is available under "define/models/soot-parameters/soot-process-parameters"

@dnwillia-work @PetroleumEngineer

**Comment by PetroleumEngineer** [comment]
what should i give to it ?, i tried to type child_names or get_attrbs but id give me error. i need to specify, per exponential factor and activation energy.
![image](https://github.com/ansys/pyfluent/assets/86235779/145bbde1-d273-4a5a-9423-bb62b596d839)

**Comment by PetroleumEngineer** [comment]
I tried for a few hours to do it through materials change-create. in this section you can define material create reactions and mechanisms. in one of the steps it asks if i want to include backward reactions. I choose 'yes', I was expecting that it will allow me to setup Arrhenius rate parameters after choosing 'yes'. but it does nothing and the next step is that it saves new mixture.
![image](https://github.com/ansys/pyfluent/assets/86235779/63c7b8ab-d1b0-4b67-aad2-de022f5c3632)

**Comment by gyeole** [comment]
Message from species team :-  "I do not think what he is looking to specify is in the soot section; what he/she is trying to do seems to be correct, but the option to do this seems to be missing from the TUI; specifically if backward reaction is selected, the prompt for whether the parameters can be explicitly specified is missing; if the answer to the prompt is yes, it should ask the same set of parameters (activation energy, pre-exponential, etc) for defining the backward reaction;"

"for backward reactions, user should not need to specify any additional information. The backward rates will be calculated from standard state entropy. "

**Comment by dnwillia-work** [comment]
OK, so the backward rates are not even available in the TUI.  I guess that must be an oversight.

**Comment by PetroleumEngineer** [comment]
I hope it will be added in the upcoming future

**Comment by prmukherj** [comment]
Moved to: https://github.com/ansys-internal/fluent-settings-api/issues/32
